File: cpp/test/DynamicKey4_test.cpp
#include "../src/DynamicKey4.h"
#include <gtest/gtest.h>
#include <string>
#include <stdint.h>

class DynamicKey4_test : public testing::Test
{
protected:
    virtual void SetUp(){}
    virtual void TearDown(){}

public:
    void test_DynamicKey4();

    void test_RecordTicket();
};

void DynamicKey4_test::test_RecordTicket(){
    auto staticKey  = "970ca35de60c44645bbae8a215061b33";
    auto  signKey   = "5cfd2fd1755d40ecb72977518be15d3b";
    auto channelName= "7d72365eb983485397e3e3f9d460bdda";
    auto  unixTs = 1446455472;
    auto  randomInt = 58964981;
    uint32_t uid=2882341273u;
    auto  expiredTs=1446455471;
    std::string result = agora::tools::DynamicKey4::generateRecordingKey(staticKey, signKey, channelName, unixTs, randomInt, uid, expiredTs);
    EXPECT_EQ(result, "004e0c24ac56aae05229a6d9389860a1a0e25e56da8970ca35de60c44645bbae8a215061b3314464554720383bbf51446455471");
}

void DynamicKey4_test::test_DynamicKey4(){
    auto staticKey  = "970ca35de60c44645bbae8a215061b33";
    auto  signKey   = "5cfd2fd1755d40ecb72977518be15d3b";
    auto channelName= "7d72365eb983485397e3e3f9d460bdda";
    auto  unixTs = 1446455472;
    auto  randomInt = 58964981;
    uint32_t uid=2882341273u;
    auto  expiredTs=1446455471;
    std::string result = agora::tools::DynamicKey4::generateMediaChannelKey(staticKey, signKey, channelName, unixTs, randomInt, uid, expiredTs);
    EXPECT_EQ(result, "004d0ec5ee3179c964fe7c0485c045541de6bff332b970ca35de60c44645bbae8a215061b3314464554720383bbf51446455471");
}

TEST_F(DynamicKey4_test, test_RecordTicket)
{
    test_RecordTicket();
}
TEST_F(DynamicKey4_test, test_DynamicKey4)
{
    test_DynamicKey4();
}

File: python/test/DynamicKey4Test.py
import sys
import unittest
import os
import time
from random import randint

sys.path.append(os.path.join(os.path.dirname(__file__), '../src'))
from DynamicKey4 import generateRecordingKey
from DynamicKey4 import generateMediaChannelKey

statickey   = "970ca35de60c44645bbae8a215061b33"
signkey     = "5cfd2fd1755d40ecb72977518be15d3b"
channelname = "7d72365eb983485397e3e3f9d460bdda"
unixts = 1463199341
uid = 2882341273
randomint = -252347445
expiredts = 0

class DynamicKeyTest(unittest.TestCase):
    def test_recording(self):
        expected = "004f0386ad262b31103ec737f06c303f778c2410fed970ca35de60c44645bbae8a215061b331463199341f0f57bcb0000000000";
        actual = generateRecordingKey(statickey, signkey, channelname, unixts, randomint, uid, expiredts);
        print actual
        self.assertEqual(expected, actual)

    def test_mediachannel(self):
        expected = "0043d8758ab4f38a56400c7bbfaae66d7ff4b5bea26970ca35de60c44645bbae8a215061b331463199341f0f57bcb0000000000";
        actual = generateMediaChannelKey(statickey, signkey, channelname, unixts, randomint, uid, expiredts);
        print actual
        self.assertEqual(expected, actual)

if __name__ == "__main__":
    unittest.main()
